### PR TITLE
[App] Cold start proxy in autoscaler

### PR DESCRIPTION
**Pull request by hhsecond**
## What does this PR do?

Cold start proxy introduces a Proxy Handler parameter into the AutoScaler to which the requests are forwarded if the main workers are not available. 

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]
LGTM!

File: src/lightning_app/components/serve/auto_scaler.py [hunk 1]
```diff
@@ -196,33 +256,51 @@ async def consumer(self):
...
-
             if request_id in self._responses:
                 result = self._responses[request_id]
                 del self._responses[request_id]
-                _raise_granular_exception(result)
+                _maybe_raise_granular_exception(result)
                 return result
 
+    def _has_processing_capacity(self):
+        """this function checks if currently have processing capacity for one more request or not.
```
Comment (tchaton) [inline src/lightning_app/components/serve/auto_scaler.py:308]:
```suggestion
        """This function checks if we have processing capacity for one more request or not.
```

File: tests/tests_app/components/serve/test_auto_scaler.py [hunk 2]
```diff
@@ -173,3 +176,55 @@ def test_autoscaler_scale_down(monkeypatch):
     auto_scaler.autoscale()
     auto_scaler.scale.assert_called_once()
     auto_scaler.remove_work.assert_called_once()
+
+
+class TestLoadBalancerProcessRequest:
```
Comment (tchaton) [inline tests/tests_app/components/serve/test_auto_scaler.py:181]: No need for a class ;)

**Review by tchaton — APPROVED** [review]
LGTM !

File: tests/tests_app/components/serve/test_auto_scaler.py (hunk 2 above)
Comment (hhsecond) [reply tests/tests_app/components/serve/test_auto_scaler.py:181]: Just grouping simiar tests together. Easy to run them together when there are changes to process request

**Pull request merged**